### PR TITLE
CTL Revision

### PR DESCRIPTION
**Pull request by ophthal**
Revision of Contact Lens Panel.
Add list of prev Rx’s icon
Improve UI for CTL Rx creation and dispense modals.
Add fields to record dispensed # of units

Signed-off-by: ophthal <magauran@ophthal.org>

More formatting changes coming...

### CODE REVIEW

File: interface/forms/eye_mag/SpectacleRx.php [hunk 1]
```diff
@@ -564,27 +598,39 @@ class="bold underline left"><?php echo xlt('Right Lens'); ?></u></td>
...
+                                        <td><?php echo text($row['CTLODVA']); ?></td>
+                                    </tr>
+                                    <tr>
+                                        <td colspan="4" class="right bold">
+                                            <?php echo xlt('Brand'); ?>: <br />
+                                            <?php echo xlt('Quantity'); ?>: <br />
+                                            <?php echo xlt('Supplier'); ?>: </td>
+                                        <td colspan="4" class="left" style="padding-left:10px;"><?php echo text($row['CTLBRANDOD']); ?>
+                                            <?php if (!empty($row['CTLMANUFACTUREROD'])) { echo "(".$row['CTLMANUFACTUREROD'].")";} ?><br />
+                                            <?php echo $row['CTLODQUANTITY']; ?><br />
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: need to use text() in above 2 lines

File: interface/forms/eye_mag/SpectacleRx.php [hunk 2]
```diff
@@ -594,9 +640,32 @@ class="bold underline left"><?php echo xlt('Right Lens'); ?></u></td>
...
+                                        <td><?php echo text($row['CTLOSVA']); ?></td>
+                                    </tr>
+                                    <tr>
+                                        <td colspan="4" class="right bold">
+                                            <?php echo xlt('Brand'); ?>: <br />
+                                            <?php echo xlt('Quantity'); ?>: <br />
+                                            <?php echo xlt('Supplier'); ?>: </td>
+                                        <td colspan="4" class="left" style="padding-left:10px;"><?php echo text($row['CTLBRANDOS']); ?>
+                                            <?php if (!empty($row['CTLMANUFACTUREROS'])) { echo "(".$row['CTLMANUFACTUREROS'].")";} ?><br />
+                                            <?php echo $row['CTLOSQUANTITY']; ?><br />
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: need to use text() in above 2 lines

File: interface/forms/eye_mag/SpectacleRx.php [hunk 3]
```diff
@@ -824,6 +896,19 @@ function reverse_cylinder() {
 <body>
 <?php echo report_header($pid, "web");  ?>
 <br/><br/>
+<?php
+    $row['REFDATE'] = oeFormatShortDate($data['date']);
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: You should not be converted to the localized date here. I think that will break the strtotime calls below. Converting it to localized date after the conversion as you do below makes more sense.

File: interface/forms/eye_mag/SpectacleRx.php [hunk 4]
```diff
@@ -1055,22 +1139,20 @@ class="prism"
...
                                 <td class="right bold underline"><?php echo xlt('Right Lens'); ?></td>
                             </tr>
-                            <tr>
-                                <td colspan="2" class="right bold"><?php echo xlt('Manufacturer'); ?>:</td>
-                                <td colspan="2" class="left"><?php echo text($CTLMANUFACTUREROD); ?></td>
-                                <td class="right bold"><?php echo xlt('Brand'); ?>:</td>
-                                <td colspan="2" class="left"><?php echo text($CTLBRANDOD); ?></td>
+                           <tr>
+                                <td colspan="2" class="right bold text-uppercase"><?php echo xlt('Brand'); ?>:</td>
+                                <td colspan="4" class="left"><?php echo text($CTLBRANDOD); ?> <?php if ($CTLMANUFACTUREROD) { echo "(".$CTLMANUFACTUREROD.")";} ?></td>
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: text() the echo of $CTLMANUFACTUREROD

File: interface/forms/eye_mag/SpectacleRx.php [hunk 5]
```diff
@@ -1094,22 +1175,23 @@ class="prism"
...
                                 </td>
-                                <td><input type=text id="CTLODVA" name="CTLODVA" value="<?php echo attr($ODVA); ?>">
-                                </td>
                             </tr>
                             <tr>
-                                <td colspan="8">
-                                    <hr/>
+                                <td colspan="2" class="right bold text-uppercase">
+                                    <?php echo xlt('Quantity:'); ?>
+                                <td colspan="4" class="left"><input id="CTLODQUANTITY" name="CTLODQUANTITY" value="<?php echo $CTLODQUANTITY; ?>" type="text" class="left" /></td></td>
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: attr() around $CTLODQUANTITY

File: interface/forms/eye_mag/SpectacleRx.php [hunk 6]
```diff
@@ -1094,22 +1175,23 @@ class="prism"
...
-                                <td class="right bold underline"><?php echo xlt('Left Lens'); ?></td>
+                                <td class="right bold large underline"><?php echo xlt('Left Lens'); ?></td>
                             </tr>
                             <tr>
-                                <td colspan="2" class="right bold"><?php echo xlt('Manufacturer'); ?>:</td>
-                                <td colspan="2" class="left"><?php echo text($CTLMANUFACTUREROS); ?></td>
-                                <td class="right bold"><?php echo xlt('Brand'); ?>:</td>
-                                <td colspan="2" class="left"><?php echo text($CTLBRANDOS); ?></td>
+                                <td colspan="2" class="right bold text-uppercase"><?php echo xlt('Brand'); ?>:</td>
+                                <td colspan="4" class="left"><?php echo text($CTLBRANDOS); ?> <?php if ($CTLMANUFACTUREROS) { echo "(".$CTLMANUFACTUREROS.")";} ?></td>
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: text() the echo of $CTLMANUFACTUREROD

File: interface/forms/eye_mag/SpectacleRx.php [hunk 7]
```diff
@@ -1133,20 +1214,24 @@ class="prism"
...
                                 <td><input type=text id="CTLOSADD" name="CTLOSADD" value="<?php echo attr($OSADD); ?>">
                                 </td>
-                                <td><input type=text id="CTLOSVA" name="CTLOSVA" value="<?php echo attr($OSVA); ?>">
-                                </td>
                             </tr>
-                            <?php if ($COMMENTS > '') { ?>
+                            <tr>
+                                <td colspan="2" class="right bold text-uppercase">
+                                    <?php echo xlt('Quantity:'); ?>
+                                <td colspan="4" class="left"><input id="CTLOSQUANTITY" name="CTLOSQUANTITY" value="<?php echo $CTLOSQUANTITY; ?>" type="text" class="left" /></td></td>
```
Comment (bradymiller) [inline interface/forms/eye_mag/SpectacleRx.php]: attr() around $CTLOSQUANTITY

File: sql/5_0_1-to-5_0_2_upgrade.sql [hunk 8]
```diff
@@ -632,3 +632,8 @@ ALTER TABLE `users_secure` ADD COLUMN `login_work_area` text;
 #IfNotColumnType onsite_messages sender_id VARCHAR(64)
 ALTER TABLE `onsite_messages` CHANGE `sender_id` `sender_id` VARCHAR(64) NULL COMMENT 'who sent id';
 #EndIf
+
+#IfMissingColumn form_eye_mag_dispense status
```
Comment (bradymiller) [inline sql/5_0_1-to-5_0_2_upgrade.sql]: Should have 2 separate ifs here:
```
#IfMissingColumn form_eye_mag_dispense CTLODQUANTITY
ALTER TABLE `form_eye_mag_dispense` ADD COLUMN `CTLODQUANTITY` varchar(255) DEFAULT NULL;
#EndIf

#IfMissingColumn form_eye_mag_dispense CTLOSQUANTITY
ALTER TABLE `form_eye_mag_dispense` ADD COLUMN `CTLOSQUANTITY` varchar(255) DEFAULT NULL;
#EndIf
```

**Comment by bradymiller** [comment]
code review is complete (just some minor issues were noted)

**Comment by ophthal** [comment]
This should cover it.

File: sql/5_0_1-to-5_0_2_upgrade.sql (hunk 8 above)
Comment (bradymiller) [reply sql/5_0_1-to-5_0_2_upgrade.sql]: will fix above after bring into codebase

**Comment by bradymiller** [comment]
hi @ophthal ,
Thanks for the improvements! I just brought this into the codebase.
-brady

**Pull request merged**